### PR TITLE
Add audio codec models

### PR DESCRIPTION
**Pull request by nithinraok**
# What does this PR do ?

Add NeMo Audio Codec Models to the collection 

**Collection**: [TTS]

# Changelog 
- Add audio codec 16kHz config
- Add NGC model to pretrained list

# Usage

### Automatically load the model from NGC

```python
from  nemo.collections.tts.models  import  AudioCodecModel
nemo_codec_model  =  AudioCodecModel.from_pretrained('audio_codec_16khz_small')
```

### Getting discrete tokens from Audio

```python
import librosa
import torch
audio,sr = librosa.load("<path_to_audio>", sr=16000)
device = 'cuda' if torch.cuda.is_available() else 'cpu'
audio_tensor = torch.from_numpy(audio).unsqueeze(dim=0).to(device)
audio_len = torch.tensor([audio_tensor[0].shape[0]]).to(device)
nemo_codec_model.freeze()
encoded_tokens, encoded_len = nemo_codec_model.encode(audio=audio_tensor, audio_len=audio_len)
```

### Reconstructing audio from discrete tokens

```python
reconstructed_audio = nemo_codec_model.decode(tokens=encoded_tokens, tokens_len=encoded_len)
```

### Listen to audio

```python
import soundfile as sf
output_audio  =  reconstructed_audio.cpu().numpy()[0]
sf.write(<path_to_output_audio>,  output_audio,  samplerate=16000)
```

# Jenkins CI
To run Jenkins, a NeMo User with write access must comment `jenkins` on the PR.

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [x] New Feature
- [ ] Bugfix
- [ ] Documentation

### CODE REVIEW

File: nemo/collections/tts/models/audio_codec.py [hunk 1]
```diff
@@ -643,4 +643,13 @@ def configure_callbacks(self):
 
     @classmethod
     def list_available_models(cls) -> List[PretrainedModelInfo]:
-        return []
+        models = []
+
+        model = PretrainedModelInfo(
+            pretrained_model_name="audio_codec_small_16khz",
+            location="https://api.ngc.nvidia.com/v2/models/nvidia/nemo/audio_codec_small_16khz/versions/v1/files/audio_codec_small_16khz.nemo",
+            description="For details about this model please refer to the model card: https://ngc.nvidia.com/catalog/models/nvidia:nemo:audio_codec_small_16khz",
```
[the remark this answers is not in the document]
Comment (nithinraok) [reply nemo/collections/tts/models/audio_codec.py:651]: why not? any other suggestions?

[the remark this answers is not in the document]
Comment (nithinraok) [reply nemo/collections/tts/models/audio_codec.py:651]: Oh my bad sorry, now I understood your comment. will fix it. Waiting for the model approval.

[the remark this answers is not in the document]
Comment (rlangman) [reply nemo/collections/tts/models/audio_codec.py:651]: `audio_codec_16khz_small` makes more sense to me.

But for 22.05khz and 44.1khz models I cannot see a good way to put the decimal in it. For those should we abbreviate like `audio_codec_22khz_small` which is slightly inaccurate, or do we want to expand everything like `audio_codec_16000hz_small`?

**Comment by nithinraok** [comment]
jenkins

**Comment by nithinraok** [comment]
jenkins

**Comment by nithinraok** [comment]
jenkins

**Pull request merged**